src/kudu/consensus/leader_election.cc

```diff
@@ -960,24 +955,16 @@ std::pair<bool, bool> FlexibleVoteCounter::ComputeElectionResultFromVotingHistor
       case PotentialNextLeadersResponse::ALL_INTERMEDIATE_TERMS_SCANNED: {
         VLOG_WITH_PREFIX(3)
             << "All intermediate terms since the last known leader: "
-            << last_known_leader.uuid() << " in term: "
-            << last_known_leader.election_term() << " were explored. "
+            << last_known_leader.uuid()
+            << " in term: " << last_known_leader.election_term()
+            << " were explored. "
             << "Current election term: " << election_term_
             << "Potential leader regions: "
-            << JoinStringsIterator(
-                r.potential_leader_regions.begin(),
-                r.potential_leader_regions.end(), ", ");
-        std::pair<bool, bool> result =
-            IsMajoritySatisfiedInPotentialLeaderRegions(
-                r.potential_leader_regions);
-        if (FLAGS_srd_strict_leader_election_quorum) {
-          std::pair<bool, bool> majority_result =
-              IsMajoritySatisfiedInMajorityOfRegions();
-          return std::make_pair<>(
-              result.first && majority_result.first,
-              result.second && majority_result.second);
-        }
-        return result;
+            << JoinStringsIterator(r.potential_leader_regions.begin(),
+                                   r.potential_leader_regions.end(), ", ");
+
+        return AreMajoritiesSatisfied(r.potential_leader_regions,
+                                      candidate_region);
```

Comment:
We're only checking special majorities after all terms are scanned and not when we're populating potential leaders.